File: ios/protocols/RNHMSProvider.swift
import Foundation
import Altcraft

@objcMembers
final class RNHMSProvider: NSObject, HMSInterface {
  private let get: (@escaping (String?) -> Void) -> Void
  private let del: (@escaping (Bool) -> Void) -> Void

  init(
    get: @escaping (@escaping (String?) -> Void) -> Void,
    del: @escaping (@escaping (Bool) -> Void) -> Void
  ) {
    self.get = get
    self.del = del
  }

  func getToken(completion: @escaping (String?) -> Void) {
    get(completion)
  }

  func deleteToken(completion: @escaping (Bool) -> Void) {
    del(completion)
  }
}

File: android/src/main/java/com/altcraft/sdk/interfaces/RnHMSProvider.kt
package com.altcraft.sdk.rn.providers

//  Created by Andrey Pogodin.
//
//  Copyright © 2025 Altcraft. All rights reserved.

import android.content.Context
import androidx.annotation.Keep
import com.altcraft.sdk.interfaces.HMSInterface

@Keep
object RnHMSProvider : HMSInterface {

    @Volatile
    private var token: String? = null

    override suspend fun getToken(context: Context): String? = token


    override suspend fun deleteToken(context: Context, complete: (Boolean) -> Unit) {
        token = null
        complete(true)
    }

    fun setToken(newToken: String?) {
        token = newToken
    }

    fun clear() {
        token = null
    }
}
